### PR TITLE
mcp: add tests for UnmarshalJSON handling of nil Content pointers

### PR DESCRIPTION
**Pull request by MrGossett**
Check that UnmarshalJSON methods for Content don't panic on nil.

- Introduces a new test file `content_nil_test.go` which verifies that `UnmarshalJSON` methods for various `Content` types do not panic when unmarshaling onto `nil` pointers.
- Adds a `nil` check in `contentFromWire` function to guard against a `nil` `wire.Content` parameter.
- Tests cover different scenarios, including valid and invalid content types, as well as cases with empty or missing content fields.

For [modelcontextprotocol#205](https://github.com/modelcontextprotocol/go-sdk/issues/205)

### CODE REVIEW

File: mcp/content_nil_test.go [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+				}
+			}()
+
+			err := json.Unmarshal([]byte(tt.json), tt.content)
+			if err != nil {
+				t.Errorf("UnmarshalJSON failed: %v", err)
+			}
+
+			// Verify that the Content field was properly populated
+			switch v := tt.content.(type) {
```
Comment (findleyr) [inline mcp/content_nil_test.go]: I think you can simplify this significantly using cmp.Diff: just compare the expected and actual value.
https://pkg.go.dev/github.com/google/go-cmp/cmp is already required by this module.

File: mcp/content.go [hunk 2]
```diff
@@ -252,6 +252,9 @@ func contentsFromWire(wires []*wireContent, allow map[string]bool) ([]Content, e
 }
 
 func contentFromWire(wire *wireContent, allow map[string]bool) (Content, error) {
+	if wire == nil {
```
Comment (findleyr) [inline mcp/content.go:255]: The test failed without this change, right?

Comment (MrGossett) [reply mcp/content.go:255]: That's correct. One of the tests that I added failed without this change.

The test is unmarshaling `{"model":"test","role":"user"}` (missing `"content"`), which causes the `*wireContent` parameter here to be nil.

Below is the test output without this change:

```text
$ go test ./mcp   
--- FAIL: TestContentUnmarshalNilWithEmptyContent (0.00s)
    --- FAIL: TestContentUnmarshalNilWithEmptyContent/Missing_Content_field (0.00s)
        content_nil_test.go:170: UnmarshalJSON panicked: runtime error: invalid memory address or nil pointer dereference
FAIL
FAIL    github.com/modelcontextprotocol/go-sdk/mcp      7.532s
FAIL
```

File: mcp/content_nil_test.go (hunk 1 above)
Comment (MrGossett) [reply mcp/content_nil_test.go]: Done in a56a3cc

**Review by findleyr — APPROVED** [review]
Thanks!
